File: backend/app/schemas/case.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import CaseStatus
from app.schemas.case_taking import CaseTakingRead
from app.schemas.consent import ConsentRead
from app.schemas.document import DocumentRead
from app.schemas.patient import PatientIdentification, PatientRead


class CaseCreate(BaseModel):
    language: str = "en"
    patient: PatientIdentification | None = None


class CaseRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    patient_id: int
    language: str
    status: CaseStatus
    created_at: datetime
    submitted_at: datetime | None


class CaseSummary(BaseModel):
    id: uuid.UUID
    language: str
    status: CaseStatus
    created_at: datetime
    submitted_at: datetime | None
    patient: PatientRead
    consents: list[ConsentRead]
    case_taking: CaseTakingRead | None
    documents: list[DocumentRead]

File: frontend/src/types/case.ts
export type CaseStatus = 'draft' | 'submitted' | 'verified';
export type Sex = 'male' | 'female' | 'other';
export type ConsentMethod = 'digital' | 'verbal' | 'written';

export interface PatientIdentification {
  full_name: string;
  dob: string; // YYYY-MM-DD
  sex: Sex;
  phone?: string | null;
  gov_id_type?: string | null;
  gov_id_ref?: string | null;
}

export interface PatientRead extends PatientIdentification {
  id: number;
}

export interface CaseCreate {
  language?: string;
  patient?: PatientIdentification | null;
}

export interface ConsentUpdate {
  consent_type: string;
  granted: boolean;
  method: ConsentMethod;
}

export interface ConsentRead {
  id: number;
  consent_type: string;
  granted: boolean;
  granted_at: string | null;
  method: ConsentMethod;
}

export interface CaseTakingUpdate {
  chief_complaint?: string | null;
  history?: string | null;
  symptoms?: Record<string, any> | null;
  vitals?: Record<string, any> | null;
}

export interface CaseTakingRead {
  id: number;
  chief_complaint: string | null;
  history: string | null;
  symptoms: Record<string, any> | null;
  vitals: Record<string, any> | null;
}

export interface DocumentRead {
  id: number;
  filename: string;
  mime_type: string;
  uploaded_at: string;
}

export interface CaseRead {
  id: string; // UUID
  patient_id: number;
  language: string;
  status: CaseStatus;
  created_at: string;
  submitted_at: string | null;
}

export interface CaseSummary {
  id: string; // UUID
  language: string;
  status: CaseStatus;
  created_at: string;
  submitted_at: string | null;
  patient: PatientRead;
  consents: ConsentRead[];
  case_taking: CaseTakingRead | null;
  documents: DocumentRead[];
}
